python/common/org/python/types/List.java

```diff
@@ -661,6 +661,27 @@ public void __delitem__(org.python.Object index) {
             args = {"other"}
     )
     public org.python.Object extend(org.python.Object other) {
+        if (other instanceof org.python.types.List) {
+            this.value.addAll(((org.python.types.List) other).value);
+        } else if (other instanceof org.python.types.FrozenSet) {
+            this.value.addAll(((org.python.types.FrozenSet) other).value);
+        } else if (other instanceof org.python.types.Set) {
+            this.value.addAll(((org.python.types.Set) other).value);
+        } else if (other instanceof org.python.types.Tuple) {
+            this.value.addAll(((org.python.types.Tuple) other).value);
+        } else if (other instanceof org.python.types.Dict) {
+            this.value.addAll(((org.python.types.Dict) other).value.keySet());
```

Comment:
I wonder whether the above can be vastly simplified with a generic method…

Reply:
I will look into that.